spec/Strategy/MockClientStrategySpec.php

```diff
@@ -24,6 +25,13 @@ function it_should_return_the_mock_client(DiscoveryStrategy $strategy)
         $candidates->shouldHaveCount(1);
     }
 
+    function it_should_return_the_mock_client_for_psr_interface(DiscoveryStrategy $strategy)
+    {
+        $candidates = $this->getCandidates(ClientInterface::class);
```

Comment:
could we do a class_exists here and pick the old or the new client interface to also be able to test this with httplug 1?

Reply:
The specification is rewritten with `class_implements`, so that we can cover all the implemented interfaces for now and in the future. `class_exists` will not do the job for us, since this specification will depend on the version of our HttpClient interface from HTTPlug.